benches/benches/bevy_ecs/commands.rs

```diff
@@ -76,6 +79,58 @@ fn spawn_commands(criterion: &mut Criterion) {
     group.finish();
 }
 
+#[derive(Default)]
+struct Matrix([[f32; 4]; 4]);
+
+#[derive(Default)]
+struct Vec3([f32; 3]);
+
+fn insert_commands(criterion: &mut Criterion) {
+    let mut group = criterion.benchmark_group("insert_commands");
+    group.warm_up_time(std::time::Duration::from_millis(500));
+    group.measurement_time(std::time::Duration::from_secs(4));
+
+    let entity_count = 10_000;
+    group.bench_function(format!("insert"), |bencher| {
+        let mut world = World::default();
+        let mut command_queue = CommandQueue::default();
+        let mut entities = Vec::new();
+        for i in 0..entity_count {
+            entities.push(world.spawn().id());
+        }
+
+        bencher.iter(|| {
+            let mut commands = Commands::new(&mut command_queue, &world);
+            for entity in entities.iter() {
+                commands.entity(*entity).insert_bundle((Matrix::default(), Vec3::default()));
+            }
+            drop(commands);
+            command_queue.apply(&mut world);
+        });
+    });
+    group.bench_function(format!("insert_batch"), |bencher| {
+        let mut world = World::default();
+        let mut command_queue = CommandQueue::default();
+        let mut entities = Vec::new();
+        for i in 0..entity_count {
+            entities.push(world.spawn().id());
+        }
+
+        bencher.iter(|| {
+            let mut commands = Commands::new(&mut command_queue, &world);
+            let mut values = Vec::with_capacity(entity_count);
+            for entity in entities.iter() {
+                values.push((*entity, (Matrix::default(), Vec3::default())));
+            }
+            commands.insert_or_spawn_batch(values);
+            drop(commands);
```

Comment:
Why do we need an explicit `drop` here?

Reply:
the alternative is to use a scope to ensure Commands is dropped (which releases the CommandQueue borrow).